dot/sync/chain_sync.go

```diff
@@ -536,15 +546,38 @@ func (cs *chainSync) requestMaxBlocksFrom(bestBlockHeader *types.Header, origin
 		}
 	}
 
-	resultsQueue := cs.workerPool.submitRequests(requests)
-	err := cs.handleWorkersResults(resultsQueue, origin, startRequestAt, expectedAmountOfBlocks)
+	resultsQueue, err := cs.submitRequests(requests)
+	if err != nil {
+		return err
```

Comment:
would suggest to add a context: `fmt.Errorf("submitting request: %w", err)`

Reply:
i am adding this to the helper functions so this will be applied on all calls to submit requests